commands/command_filter.go

```diff
@@ -0,0 +1,221 @@
+package commands
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/github/git-lfs/config"
+	"github.com/github/git-lfs/errutil"
+	"github.com/github/git-lfs/lfs"
+	"github.com/github/git-lfs/progress"
+	"github.com/spf13/cobra"
+)
+
+const (
+	FilterDriverVersion = 1
+)
+
+type InputFileHdr struct {
+	FileName string
+	FileLen  uint64
+}
+
+func (h *InputFileHdr) Read(r *bufio.Reader) error {
+	fileName, err := r.ReadString('\n')
+	if err == io.EOF {
+		return err
+	} else if err != nil {
+		Panic(err, "Error reading filter command.")
+	}
+	h.FileName = strings.TrimSuffix(fileName, "\n")
+
+	fileLenStr, err := r.ReadString('\n')
+	if err == io.EOF {
+		return err
+	} else if err != nil {
+		Panic(err, "Error reading input data length.")
+	}
+	fileLenStr = strings.TrimSuffix(fileLenStr, "\n")
+	fileLen, err := strconv.ParseUint(fileLenStr, 10, 64)
+	if err != nil {
+		Panic(err, fmt.Sprintf("Error parsing input data length '%s'.", fileLenStr))
+	}
+	h.FileLen = fileLen
+
+	// Panic(err, fmt.Sprintf("Errorddddddddddddgth '%d'.", fileLen))
+
+	return nil
+}
+
+var (
+	filterSmudgeSkip = false
+	filterCmd        = &cobra.Command{
+		Use: "filter",
+		Run: filterCommand,
+	}
+)
+
+func clean(reader io.Reader, fileName string) ([]byte, error) {
+	var cb progress.CopyCallback
+	var file *os.File
+	var fileSize int64
+	if len(fileName) > 0 {
+		stat, err := os.Stat(fileName)
+		if err == nil && stat != nil {
+			fileSize = stat.Size()
+
+			localCb, localFile, err := lfs.CopyCallbackFile("clean", fileName, 1, 1)
+			if err != nil {
+				Error(err.Error())
+			} else {
+				cb = localCb
+				file = localFile
+			}
+		}
+	}
+
+	cleaned, err := lfs.PointerClean(reader, fileName, fileSize, cb)
+	if file != nil {
+		file.Close()
+	}
+
+	if cleaned != nil {
+		defer cleaned.Teardown()
+	}
+
+	if errutil.IsCleanPointerError(err) {
+		// TODO: What is happening here? (triggered by ./test/test-clone.sh)
+		return errutil.ErrorGetContext(err, "bytes").([]byte), nil
```

Comment:
This error comes up if it's trying to run an LFS pointer through `git lfs clean`, instead of the actual contents of the file.